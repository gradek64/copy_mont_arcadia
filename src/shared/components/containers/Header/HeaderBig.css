/** @define HeaderBig */

.HeaderBig {
  display: none;
  background-color: $header-big-background;
  max-width: $header-big-max-width;
  margin: $header-big-margin;
  padding: $header-big-padding;

  &-content {
    background-color: $header-background;
    height: $header-big-height;
    position: relative;
    width: 100%;
    max-width: $laptop-end-width;
    margin: 0 auto;
    display: flex;
    align-items: center;
    justify-content: center;
    padding-top: $header-big-content-padding-top;
  }

  &-brandLogo {
    height: $header-big-brand-logo-height;
  }

  &-centre {
    flex: $header-big-centre-flex-grow;
  }

  &-left {
    height: 100%;
    flex: $header-big-left-flex-grow;
    display: flex;
    align-items: $header-big-left-align-items;
    justify-content: $header-big-left-justify-content;
  }

  &-right {
    height: 100%;
    flex: $header-big-right-flex-grow;
    display: flex;
    align-items: $header-big-right-elements-alignment;
    justify-content: flex-end;
  }
}

@media $min-tablet {
  .HeaderBig {
    display: block;
    min-height: $header-big-min-height;
  }
}
