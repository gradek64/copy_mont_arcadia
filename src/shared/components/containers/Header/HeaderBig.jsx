/*
 * This is a dedicated component that we are going to use for all the widths bigger than 992.
 * We have chosen to create a brand new header component to make as less likely as possible the possibility to
 * break the mobile header.
 */

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import * as headers from './'

@connect((state) => ({
  brandName: state.config.brandName,
  shoppingBagTotalItems: state.shoppingBag.totalItems,
  logoVersion: state.config.logoVersion,
  featureMegaNav: state.features.status.FEATURE_MEGA_NAV,
  region: state.config.region,
  sticky: state.pageHeader.sticky,
}))
class HeaderBig extends Component {
  static propTypes = {
    shoppingBagTotalItems: PropTypes.number.isRequired,
    brandName: PropTypes.string,
    logoVersion: PropTypes.string,
    featureMegaNav: PropTypes.bool,
    region: PropTypes.string,
    sticky: PropTypes.bool,
  }

  render() {
    const {
      region,
      brandName,
      shoppingBagTotalItems,
      logoVersion,
      featureMegaNav,
      sticky,
    } = this.props
    const HeaderComponent = headers[brandName]
    return (
      <div className="HeaderBig">
        <HeaderComponent
          sticky={sticky}
          brandName={brandName}
          contentWrapper="HeaderBig-content"
          leftCol="HeaderBig-left"
          rightCol="HeaderBig-right"
          centreCol="HeaderBig-centre"
          logoVersion={logoVersion}
          shoppingBagTotalItems={shoppingBagTotalItems}
          featureMegaNav={featureMegaNav}
          region={region}
        />
      </div>
    )
  }
}

export default HeaderBig
