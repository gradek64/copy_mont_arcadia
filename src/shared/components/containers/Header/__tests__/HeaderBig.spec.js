import testComponentHelper from 'test/unit/helpers/test-component'
import HeaderBig from '../HeaderBig'

describe('<HeaderBig />', () => {
  const renderComponent = testComponentHelper(HeaderBig.WrappedComponent)
  const initialProps = {
    brandName: 'topshop',
    shoppingBagTotalItems: 3,
    sticky: false,
    isFeatureStickyHeaderEnabled: false,
  }

  describe('@renders', () => {
    it('in default state', () => {
      expect(renderComponent(initialProps).getTree()).toMatchSnapshot()
    })
    it('with featureMegaNav', () => {
      expect(
        renderComponent({
          ...initialProps,
          featureMegaNav: true,
        }).getTree()
      ).toMatchSnapshot()
    })
    describe('with sticky headers feature enabled', () => {
      it('with default props', () => {
        expect(
          renderComponent({
            ...initialProps,
            isFeatureStickyHeaderEnabled: true,
          }).getTree()
        ).toMatchSnapshot()
      })
      it('with sticky=true', () => {
        expect(
          renderComponent({
            ...initialProps,
            isFeatureStickyHeaderEnabled: true,
            sticky: true,
          }).getTree()
        ).toMatchSnapshot()
      })
    })
  })
})
