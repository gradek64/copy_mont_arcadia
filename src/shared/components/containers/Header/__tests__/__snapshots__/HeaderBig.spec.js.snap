// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderBig /> @renders in default state 1`] = `
<div
  className="HeaderBig"
>
  <Connect(HeaderTopshop)
    brandName="topshop"
    centreCol="HeaderBig-centre"
    contentWrapper="HeaderBig-content"
    leftCol="HeaderBig-left"
    rightCol="HeaderBig-right"
    shoppingBagTotalItems={3}
    sticky={false}
  />
</div>
`;

exports[`<HeaderBig /> @renders with featureMegaNav 1`] = `
<div
  className="HeaderBig"
>
  <Connect(HeaderTopshop)
    brandName="topshop"
    centreCol="HeaderBig-centre"
    contentWrapper="HeaderBig-content"
    featureMegaNav={true}
    leftCol="HeaderBig-left"
    rightCol="HeaderBig-right"
    shoppingBagTotalItems={3}
    sticky={false}
  />
</div>
`;

exports[`<HeaderBig /> @renders with sticky headers feature enabled with default props 1`] = `
<div
  className="HeaderBig"
>
  <Connect(HeaderTopshop)
    brandName="topshop"
    centreCol="HeaderBig-centre"
    contentWrapper="HeaderBig-content"
    leftCol="HeaderBig-left"
    rightCol="HeaderBig-right"
    shoppingBagTotalItems={3}
    sticky={false}
  />
</div>
`;

exports[`<HeaderBig /> @renders with sticky headers feature enabled with sticky=true 1`] = `
<div
  className="HeaderBig"
>
  <Connect(HeaderTopshop)
    brandName="topshop"
    centreCol="HeaderBig-centre"
    contentWrapper="HeaderBig-content"
    leftCol="HeaderBig-left"
    rightCol="HeaderBig-right"
    shoppingBagTotalItems={3}
    sticky={true}
  />
</div>
`;
